metasrv/tests/it/grpc/metasrv_grpc_watch.rs

```diff
@@ -0,0 +1,279 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::thread;
+use std::time::Duration;
+
+use common_base::tokio;
+use common_base::tokio::sync::mpsc;
+use common_base::tokio::sync::mpsc::UnboundedReceiver;
+use common_base::tokio::sync::mpsc::UnboundedSender;
+use common_meta_api::KVApi;
+use common_meta_grpc::MetaGrpcClient;
+use common_meta_types::protobuf::event::SeqV;
+use common_meta_types::protobuf::watch_request::FilterType;
+use common_meta_types::protobuf::Event;
+use common_meta_types::protobuf::WatchRequest;
+use common_meta_types::MatchSeq;
+use common_meta_types::Operation;
+use common_meta_types::UpsertKVAction;
+
+use crate::init_meta_ut;
+
+async fn watcher_client_main(
+    addr: String,
+    watch: WatchRequest,
+    start_tx: UnboundedSender<()>,
+    mut events_rx: UnboundedReceiver<Vec<Event>>,
```

Comment:
`watcher_client_main` simulate the watcher client, `test_watch_main` simulate the client which update data to metasrv, one for watcher, one for updater.

Reply:
Put what `watcher_client_main` does in the test function body and the simple data-updater in another task.
This will get rid of these several channels for communication.